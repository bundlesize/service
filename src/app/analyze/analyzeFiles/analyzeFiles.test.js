const { analyzeFiles } = require('.')
const {
    currentBranchFileDetails,
    baseBranchFileDetails,
} = require('./analyzeFiles.mockdata')

describe('analyzeFiles', () => {
    it('analyses files correctly', () => {
        expect(
            analyzeFiles({
                baseBranchName: 'master',
                baseBranchFileDetails,
                currentBranchFileDetails,
            }),
        ).toMatchSnapshot()
    })
})
