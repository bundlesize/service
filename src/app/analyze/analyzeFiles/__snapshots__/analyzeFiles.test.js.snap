// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`analyzeFiles analyses files correctly 1`] = `
Array [
  Object {
    "error": "Failed to compare",
    "filePath": "path/to/errored/file",
    "status": "fail",
  },
  Object {
    "baseBranchSize": 0,
    "filePath": "path/to/larger/file",
    "maxSize": 50,
    "message": "60B > 50B (no compression)",
    "size": 60,
    "status": "fail",
  },
  Object {
    "baseBranchSize": 0,
    "filePath": "path/to/smaller/file",
    "maxSize": 50,
    "message": "30B < 50B (no compression)",
    "size": 30,
    "status": "pass",
  },
  Object {
    "baseBranchSize": 0,
    "filePath": "path/to/unbound/file",
    "maxSize": null,
    "message": "1KB > Infinity (no compression)",
    "size": 1024,
    "status": "fail",
  },
  Object {
    "baseBranchSize": 30,
    "filePath": "path/to/removed/file",
    "maxSize": 0,
    "message": "path/to/removed/file: File removed (30B smaller than master) (no compression)",
    "size": 0,
    "status": "removed",
  },
]
`;
